src/invariant/BucketListIsConsistentWithDatabase.cpp

```diff
@@ -90,6 +88,8 @@ BucketListIsConsistentWithDatabase::checkOnBucketApply(
     std::shared_ptr<Bucket const> bucket, uint32_t oldestLedger,
     uint32_t newestLedger)
 {
+    mApp.getLedgerStateRoot().flushCache();
```

Comment:
why do we need the call to `flushCache()` here? This function only reads from the database... I would expect writers being the only ones that need to do this (under very specific conditions)